sycl/include/CL/sycl/ONEAPI/reduction.hpp

```diff
@@ -225,6 +225,27 @@ struct known_identity_impl<BinaryOperation, AccumulatorT,
           : std::numeric_limits<AccumulatorT>::lowest();
 };
 
+#ifdef __SYCL_REDUCER_OP_EQ_CHECK_TRAIT
+#error "__SYCL_REDUCER_OP_EQ_CHECK_TRAIT must not be defined"
+#endif
+
+#define __SYCL_REDUCER_OP_EQ_CHECK_TRAIT(OpName, Op)                           \
+  template <typename, typename = void>                                         \
+  struct HasSameTypeArg##OpName##Eq : public std::false_type {};               \
+  template <typename T>                                                        \
+  struct HasSameTypeArg##OpName##Eq<                                           \
+      T, std::enable_if_t<std::is_same<                                        \
+             decltype(static_cast<T &(T::*)(const T &)>(&T::operator+=)),      \
+             T &(T::*)(const T &)>::value>> : public std::true_type {};
```

Comment:
I think a comment explaining what the SFINAE is testing here would be a good idea.

Reply:
This type trait is removed in 462c11cb89967d92cdb079bd4e9c4f69fa023c72